daemon/daemon_unix.go

```diff
@@ -375,6 +390,13 @@ func verifyPlatformContainerSettings(daemon *Daemon, hostConfig *containertypes.
 		warnings = append(warnings, "IPv4 forwarding is disabled. Networking will not work.")
 		logrus.Warnf("IPv4 forwarding is disabled. Networking will not work")
 	}
+	if hostConfig.CgroupParent != "" && daemon.usingSystemd() {
+		// CgroupParent for systemd cgroup should be named as "xxx.slice"
+		length := len(hostConfig.CgroupParent)
+		if length <= 6 || hostConfig.CgroupParent[length-6:] != ".slice" {
```

Comment:
Well, it's not a systemd bug (at least as far as I know, the systemd people appear to consider it a feature). But we're limiting Docker users on a very specific thing: "you must be inside a systemd slice". It is not correct to say that `--cgroup-parent=some.slice` will always refer to the same slice for all processes.

Just look at the paths in `/proc/<pid>/cgroup` for any process managed by systemd (including  your shell). You'll notice that the paths aren't the same and that `--cgroup-parent=some.slice` will confuse people if you're trying to put a container into a slice called `some.slice`.